tests/Transforms/convert_to_ciphertext_semantics/tensor_insert.mlir

```diff
@@ -1,49 +1,97 @@
+// CHECK-DAG:    %[[c0_i16:.*]] = arith.constant 0 : i16
+// CHECK-DAG:    %[[c7:.*]] = arith.constant 7 : i16
+// CHECK-DAG:    %[[dense1:.*]] = arith.constant dense<1> : tensor<1x16xi16>
+// CHECK-DAG:    %[[dense0:.*]] = arith.constant dense<0> : tensor<1x16xi16>
+// CHECK:    %[[res:.*]] = secret.generic(%[[arg0]]: !secret.secret<tensor<1x16xi16>>) {
+// CHECK:    ^body(%[[input:.*]]: tensor<1x16xi16>):
+
+//           The loop that implements the codegen for the masks
+// CHECK:      scf.for
+// CHECK:      scf.if
+// CHECK:      tensor.insert
+// CHECK:      tensor.insert
+// CHECK:      scf.yield
+// CHECK:      } else {
+// CHECK:      scf.yield
+// CHECK:      }
+// CHECK:      scf.yield
+
+// CHECK:      arith.muli
+// CHECK:      arith.addi
+// CHECK:    return
+
+func.func @insert_cleartext_into_secret_dynamic(%arg0: !secret.secret<tensor<8xi16>> {tensor_ext.layout = #layout}, %insertIndex: index) -> (!secret.secret<tensor<8xi16>> {tensor_ext.layout = #layout}) {
+  %c7 = arith.constant 7 : i16
+  %0 = secret.generic(%arg0 : !secret.secret<tensor<8xi16>> {tensor_ext.layout = #layout}) {
+  ^body(%input0: tensor<8xi16>):
+    %0 = tensor.insert %c7 into %input0[%insertIndex] {tensor_ext.layout = #layout} : tensor<8xi16>
+    secret.yield %0 : tensor<8xi16>
+  } -> (!secret.secret<tensor<8xi16>> {tensor_ext.layout = #layout})
+  return %0 : !secret.secret<tensor<8xi16>>
+}
+
+// CHECK:   func.func @insert_secret_into_secret(%[[tensor_secret:[^:]*]]: !secret.secret<tensor<1x16xi16>>
+// CHECK-SAME:       %[[scalar_secret:[^:]*]]: !secret.secret<tensor<1x16xi16>>
+// CHECK-DAG:    %[[c11:.*]] = arith.constant 11 : index
+// CHECK-DAG:    %[[c0_index:.*]] = arith.constant 0 : index
+// CHECK-DAG:    %[[c0_i16:.*]] = arith.constant 0 : i16
+// CHECK-DAG:    %[[c3:.*]] = arith.constant 3 : index
+// CHECK-DAG:    %[[c1:.*]] = arith.constant 1 : i16
+// CHECK-DAG:    %[[dense1:.*]] = arith.constant dense<1> : tensor<1x16xi16>
+// CHECK-DAG:    %[[dense0:.*]] = arith.constant dense<0> : tensor<1x16xi16>
+// CHECK:    %[[res:.*]] = secret.generic(%[[tensor_secret]]: !secret.secret<tensor<1x16xi16>>, %[[scalar_secret]]: !secret.secret<tensor<1x16xi16>>) {
+// CHECK:    ^body(%[[tensor_clear:.*]]: tensor<1x16xi16>, %[[scalar_clear:.*]]: tensor<1x16xi16>):
+
+// This test differs from other tests in that the scalar mask has 1's and is then multiplied with the scalar_clear
+// CHECK:      %[[scalar_mask1:.*]] = tensor.insert %[[c1]] into %[[dense0]][%[[c0_index]], %[[c3]]]
+// CHECK:      tensor.insert %[[c1]] into %[[scalar_mask1]][%[[c0_index]], %[[c11]]]
+// CHECK:      arith.muli
+// CHECK:      arith.muli
+// CHECK:      arith.addi
+// CHECK:    return %[[res]]
+func.func @insert_secret_into_secret(%arg0: !secret.secret<tensor<8xi16>> {tensor_ext.layout = #layout}, %arg1: !secret.secret<i16> {tensor_ext.layout = #scalar_layout}) -> (!secret.secret<tensor<8xi16>> {tensor_ext.layout = #layout}) {
```

Comment:
what happens when it's secret into cleartext? in the matchandrewrite code we assume that the layout would be assigned to a cleartext, so i think it would follow the same lowering as secret into secret? and then masks would cleared up on the plaintext if they can.